packages/router/src/adapters/contract/contract.ts

```diff
@@ -628,8 +731,18 @@ export const cancelRouterContract = async (
       const { event } = await cancelEvt
         .pipe(({ args }) => args.txData.transactionId === txData.transactionId)
         .waitFor(300_000);
+
+      // increment router fees
+      incrementRelayerFeesPaid(
```

Comment:
this needs to come *after* we call `getTransactionReceipt`! We have no idea if the relayer fees are paid at all until we get a receipt back indicating it executed.

Reply:
What do you mean? I'm not sure of the edgecase you're trying to address here (I think gelato just doesn't return anything if it fails)

Reply:
I think @jakekidd's logic is correct here. Unless we know the tx succeeded, we won't know that the relayer fee was taken from the contract (which only happens in a successful contract call). However, it should be at this point if this EVT didn't error.

Reply:
I mean, the stalling here comes from the `getTransactionReceipt` call. Our gelato relayer call is just an API POST request and we have no idea whether it's successful until we either get the receipt back from `getTransactionReceipt` or we timeout doing so.